### PR TITLE
Karpenter pattern

### PR DESCRIPTION
**Pull request by zjaco13**
*Issue #, if available:* https://github.com/aws-quickstart/cdk-eks-blueprints/issues/764

*Description of changes:* Add a karpenter pattern and docs, fix graviton docs

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/karpenter-construct/index.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+        const karpenterAddon = new blueprints.addons.KarpenterAddOn({
+            requirements: [
+                { key: 'node.kubernetes.io/instance-type', op: 'In', vals: ['m5.large'] },
+                { key: 'topology.kubernetes.io/zone', op: 'NotIn', vals: ['us-west-2c']},
+                { key: 'kubernetes.io/arch', op: 'In', vals: ['amd64','arm64']},
+                { key: 'karpenter.sh/capacity-type', op: 'In', vals: ['on-demand']},
+            ],
+            subnetTags: {
+                "Name": "karpenter-blueprint/karpenter-blueprint-vpc/PrivateSubnet1",
```
Comment (youngjeong46) [inline lib/karpenter-construct/index.ts:18]: let's try using variables here, i.e. ${StackID}. Also we can probably use wildcards to include all Private Subnets.

File: lib/karpenter-construct/index.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            },
+            securityGroupTags: {
+                "kubernetes.io/cluster/karpenter-blueprint": "owned",
```
Comment (youngjeong46) [inline lib/karpenter-construct/index.ts:21]: same here as above on the stackId

**Review by youngjeong46 — COMMENTED** [review]
Minor comments.

File: lib/karpenter-construct/index.ts (hunk 2 above)
Comment (zjaco13) [reply lib/karpenter-construct/index.ts:21]: When I try that for this one with `kubernetes.io/cluster/${stackID}`: "owned", I get a bunch of typescript errors

Comment (zjaco13) [reply lib/karpenter-construct/index.ts]: resolved

File: lib/karpenter-construct/index.ts (hunk 1 above)
Comment (zjaco13) [reply lib/karpenter-construct/index.ts]: resolved

**Review by youngjeong46 — COMMENTED** [review]
@zjaco13 Can you confirm this deploys as intended on your account? Then we can approve and merge.

**Comment by zjaco13** [comment]
> 

Can confirm it deploys as intended

**Review by youngjeong46 — APPROVED** [review]
LGTM.

**Pull request merged**